corehq/apps/app_manager/app_schemas/tests/test_case_properties.py

```diff
@@ -181,3 +193,15 @@ class DocTests(SimpleTestCase):
     def test_doctests(self):
         results = doctest.testmod(corehq.apps.app_manager.app_schemas.case_properties)
         self.assertEqual(results.failed, 0)
+
+
+class ReplacePropertyWithAttributesTests(SimpleTestCase):
```

Comment:
@millerdev Are these tests what you had in mind for protecting against the iteration issues? The code as it was written actually cannot generate an error (yet), because it's iterating over a set and only modifies a single attribute. So, any item that gets skipped does not matter. However, it's obviously a good idea to guard against future errors, and hopefully this conveys that idea?